cloud/app/server/telemetry.js

```diff
@@ -39,9 +42,53 @@ class TelemetryService {
       );
       this.logProcessor = logProcessor;
 
-      log.info('OpenTelemetry native logger initialized successfully');
+      // Create OTLP metrics exporter
+      const metricExporter = new OTLPMetricExporter({
+        url: 'http://otel-collector:4318/v1/metrics',
+        headers: {
+          'authorization': this.hyperDXIngestionAPIKey,
+          'content-Type': 'application/json',
+        },
+        timeoutMillis: 5000,
+      });
+
+      log.debug('Creating metrics provider...');
+      const metricReader = new PeriodicExportingMetricReader({
+        exporter: metricExporter,
+        exportIntervalMillis: 30000, // Export every 30 seconds
+        exportTimeoutMillis: 5000,
+      });
+
+      this.meterProvider = new MeterProvider({
+        readers: [metricReader],
+      });
+
+      const meter = this.meterProvider.getMeter('resourceMonitor', '1.0.0');
+
+      // Create gauges for CPU and memory metrics
+      this.cpuGauge = meter.createGauge('cpu_usage_percent', {
+        description: 'CPU usage percentage',
+        unit: '%',
+      });
```

Comment:
An exporter, a reader, a meter and then a gauge. Seems complicated. Why is that better than just sending the metrics over HTTP directly?